src/mainwindow.cpp

```diff
@@ -2624,11 +2624,13 @@ void MainWindow::mouseMoveEvent(QMouseEvent *event)
 #  endif
 
     if (m_canMoveWindow) {
-        int dx = event->globalX() - m_mousePressX;
-        int dy = event->globalY() - m_mousePressY;
-        move(dx, dy);
-
+        if (!window()->windowHandle()->startSystemMove()) {
```

Comment:
I have no idea whether it is fine to continuously call this function on all platforms. Or does `mouseMoveEvent` stop being called when this function succeeded? The documentation doesn't say anything about it, so I guess we could just test.

Alternatively, this function could be called in `MainWindow::mousePressEvent` instead, where we maybe could do `m_canMoveWindow = !window()->windowHandle()->startSystemMove()` instead of just assigning `true` (untested)